File: Java/CodingBat/alarmClock.java
// http://codingbat.com/prob/p160543

public String alarmClock(int day, boolean vacation) {
  String result = "";
  boolean isWeekday = (day >= 1 && day <= 5);
  
  if (vacation) {
    if (isWeekday) {
      result = "10:00";
    }
    else {
      result = "off";
    }
  }
  else {
    if (isWeekday) {
      result = "7:00";
    }
    else {
      result = "10:00";
    }
  }
  
  return result;
}

File: Python/CodingBat/alarm_clock.py
# http://codingbat.com/prob/p119867

def alarm_clock(day, vacation):
  is_weekday = ( day in range(1, 6) )
  
  if vacation:
    if is_weekday:
      return "10:00"
    else:
      return "off"
  else:
    if is_weekday:
      return "7:00"
    else:
      return "10:00"
